homeassistant/components/switch/rpi_gpio.py

```diff
@@ -1,69 +1,48 @@
 """
 homeassistant.components.switch.rpi_gpio
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-Allows to control the GPIO pins of a Raspberry Pi.
+Allows to configure a switch using RPi GPIO.
 
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/switch.rpi_gpio/
 """
+
 import logging
-try:
-    import RPi.GPIO as GPIO
-except ImportError:
-    GPIO = None
+import homeassistant.components.rpi_gpio as rpi_gpio
 from homeassistant.helpers.entity import ToggleEntity
-from homeassistant.const import (DEVICE_DEFAULT_NAME,
-                                 EVENT_HOMEASSISTANT_START,
-                                 EVENT_HOMEASSISTANT_STOP)
+from homeassistant.const import (DEVICE_DEFAULT_NAME)
 
 DEFAULT_INVERT_LOGIC = False
 
-REQUIREMENTS = ['RPi.GPIO==0.5.11']
+DEPENDENCIES = ['rpi_gpio']
 _LOGGER = logging.getLogger(__name__)
 
 
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices, discovery_info=None):
-    """ Sets up the Raspberry PI GPIO ports. """
-    if GPIO is None:
-        _LOGGER.error('RPi.GPIO not available. rpi_gpio ports ignored.')
-        return
-    # pylint: disable=no-member
-    GPIO.setmode(GPIO.BCM)
+    """ Sets up the Raspberry PI GPIO devices. """
 
-    switches = []
     invert_logic = config.get('invert_logic', DEFAULT_INVERT_LOGIC)
+
+    switches = []
     ports = config.get('ports')
-    for port_num, port_name in ports.items():
-        switches.append(RPiGPIOSwitch(port_name, port_num, invert_logic))
+    for port, name in ports.items():
+        switches.append(RPiGPIOSwitch(name, port, invert_logic))
     add_devices(switches)
 
-    def cleanup_gpio(event):
-        """ Stuff to do before stop home assistant. """
-        # pylint: disable=no-member
-        GPIO.cleanup()
-
-    def prepare_gpio(event):
-        """ Stuff to do when home assistant starts. """
-        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, cleanup_gpio)
-
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, prepare_gpio)
-
 
 class RPiGPIOSwitch(ToggleEntity):
-    """ Represents a port that can be toggled using Raspberry Pi GPIO. """
-
-    def __init__(self, name, gpio, invert_logic):
+    """ Represents a switch that can be toggled using Raspberry Pi GPIO. """
+    def __init__(self, name, port, invert_logic):
         self._name = name or DEVICE_DEFAULT_NAME
-        self._gpio = gpio
-        self._active_state = not invert_logic
-        self._state = not self._active_state
-        # pylint: disable=no-member
-        GPIO.setup(gpio, GPIO.OUT)
+        self._port = port
+        self._invert_logic = invert_logic
+        self._state = False
```

Comment:
The pin state will always be controlled by HA so nowhere to retrieve it from. 
Something that can be done is to add a config parameter called `initial_state`.